File: windows/.agent/skills/upsert-case/references/lifecycle-pattern.cpp
// Proper Engine Lifecycle Pattern

// Create
m_rtcEngine = createAgoraRtcEngine();

// Initialize
RtcEngineContext context;
context.appId = CConfig::GetAppId();
context.eventHandler = &m_eventHandler;
m_rtcEngine->initialize(context);

// Use
m_rtcEngine->joinChannel(token, channelName, "", 0);

// Cleanup
m_rtcEngine->leaveChannel();
m_rtcEngine->release();
m_rtcEngine = nullptr;

File: macOS/.agent/skills/upsert-case/references/lifecycle-pattern.swift
// Proper Engine Lifecycle Pattern

// Create
let config = AgoraRtcEngineConfig()
config.appId = KeyCenter.AppId
agoraKit = AgoraRtcEngineKit(config: config, delegate: self)

// Use
agoraKit.joinChannel(byToken: token, channelName: channel, info: nil, uid: 0)

// Cleanup
agoraKit.leaveChannel(nil)
agoraKit.destroy()
